src/views/Activate/ActivateCode.js

```diff
@@ -0,0 +1,150 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import * as azimuth from 'azimuth-js';
+import Maybe from 'folktale/maybe';
+import { useTicketInput } from 'components/Inputs';
+import { Grid, Input, H4, ErrorText } from 'indigo-react';
+
+import View from 'components/View';
+import { ForwardButton } from 'components/Buttons';
+import Passport from 'components/Passport';
+
+import { useHistory } from 'store/history';
+
+import * as need from 'lib/need';
+import { ROUTE_NAMES } from 'lib/routeNames';
+import { useSyncKnownPoints } from 'lib/useSyncPoints';
+import FooterButton from 'components/FooterButton';
+import Blinky from 'components/Blinky';
+import { DEFAULT_HD_PATH, walletFromMnemonic } from 'lib/wallet';
+import { useNetwork } from 'store/network';
+import { generateWallet } from 'lib/invite';
+import { generateTemporaryOwnershipWallet } from 'lib/walletgen';
+import { useActivateFlow } from './ActivateFlow';
+import { useLocalRouter } from 'lib/LocalRouter';
+
+export default function ActivateCode() {
+  const history = useHistory();
+  const { names, push } = useLocalRouter();
+  const { contracts } = useNetwork();
+  const [generalError, setGeneralError] = useState();
+  const [deriving, setDeriving] = useState(false);
+  const {
+    derivedWallet,
+    setDerivedWallet,
+    setInviteWallet,
+    derivedPoint,
+    setDerivedPoint,
+  } = useActivateFlow();
+
+  const ticketInput = useTicketInput({
+    name: 'ticket',
+    label: 'Code',
+    autoFocus: true,
+  });
+  const validTicket = ticketInput.pass;
+
+  const goToLogin = useCallback(() => history.popAndPush(ROUTE_NAMES.LOGIN), [
+    history,
+  ]);
+  const goToDisclaimer = useCallback(() => push(names.DISCLAIMER), [
+    names,
+    push,
+  ]);
+
+  const pass = derivedWallet.matchWith({
+    Nothing: () => false,
+    Just: () => true,
+  });
+
+  useEffect(() => {
+    if (validTicket) {
+      const _contracts = need.contracts(contracts);
+
+      setDeriving(true);
+      // when the ticket becomes valid, derive the point
+      (async () => {
+        const { seed } = await generateTemporaryOwnershipWallet(
+          ticketInput.data
+        );
+
+        //TODO isn't all this accessible in the ownership object?
```

Comment:
Yes! You can just `{ keys }` above. [ref](https://github.com/urbit/urbit-key-generation/blob/master/src/index.js#L172-L175) (This reads like one of my own TODOs though. (^: )